### PR TITLE
test: traces tests

### PR DESCRIPTION
**Pull request by priyabrata-stack**
### **User description**
Trace data ingestion utility (trace-ingestion.js) that generates OTLP-compliant distributed traces with various scenarios (success, database errors, auth errors, 404s)
Reusable test helper utilities following DRY principles from logs tests
6 test spec files with P0/P1/P2 prioritization covering critical paths, core functionality, and edge cases
Enhanced page object with 50+ new selectors and methods for trace interactions
Global setup integration to ingest 20 test traces before test execution

___

### **PR Type**
Tests

___

### **Description**
- Extend `TracesPage` with selectors and helper methods  

- Add global setup trace ingestion utility  

- Introduce Playwright test suites for Traces:  
  - Service Maps, Advanced Filtering, Trace Details  
  - Error Filtering, Query Editor, Search

___

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>tracesPage.js</strong><dd><code>Extend TracesPage with selectors and helpers</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-b21e724124188921fea3b88d62aef68acd31e385e1b04d3f23565f3f9e059bb3">+253/-6</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>trace-ingestion.js</strong><dd><code>Implement trace ingestion utility</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-83e4c6bffd71f42bc78f33487b04cd1613e94c6375b706274e6841efffd8dd52">+331/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td><strong>serviceMaps.spec.js</strong><dd><code>Add service maps view tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-b16eddc62227ec356faceb5c640413311d5b569747084b138bacf08427d022ca">+182/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>traceAdvancedFiltering.spec.js</strong><dd><code>Add advanced filtering test scenarios</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-3bbb700e0ea018647160af79231dc606f176fcb7e0be1164b69aeed91fc9026a">+360/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>traceDetails.spec.js</strong><dd><code>Add trace details functionality tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-b01297d7b1f713590213d42937b05c25621d12d30dd0e4144fb861db15b15810">+198/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>traceErrorFilter.spec.js</strong><dd><code>Add error filtering test suite</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-7ed3a7e2fbc68ea2786a3fee2f83d3ee33ce2ac45830e9253aeebd94d63441c2">+284/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>traceQueryEditor.spec.js</strong><dd><code>Add query editor interaction tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-0f3a8bffbd91aabd267920147a03425566a8a7c2dd74795ff15aed23a87dcead">+263/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>tracesSearch.spec.js</strong><dd><code>Add basic traces search and navigation tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-75c29494bb340e63aa8cf76bdcff572d9e860c9b4a7fa8ba3510ebc86ccdda60">+298/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>trace-test-helpers.js</strong><dd><code>Add trace test helper utilities</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-12fffc3c28d1b21cd02f87065767fcb9655ef22e2210093daaf6a67ad01f406f">+328/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>global-setup.js</strong><dd><code>Integrate trace ingestion in global setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/9650/files#diff-06323dad7214f86b41b62e121e716eeea0054bc4d950c6bcf4a0efe53894acc8">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by priyabrata-stack** [comment]
@greptile

**Comment by priyabrata-stack** [comment]
@greptile

**Comment by priyabrata-stack** [comment]
@greptile

**Comment by priyabrata-stack** [comment]
@greprile

File: tests/ui-testing/pages/tracesPages/tracesPage.js [hunk 1]
```diff
@@ -4,21 +4,80 @@ import { expect } from '@playwright/test';
...
     this.tracesMenuItem = page.locator('[data-test="menu-link-\\/traces-item"]');
 
+    // Search Bar - Tab Toggles
+    this.searchToggle = '[data-test="traces-search-toggle"]';
+    this.serviceMapsToggle = '[data-test="traces-service-maps-toggle"]';
+
+    // Search Bar - Controls
+    this.showMetricsToggle = '[data-test="traces-search-bar-show-metrics-toggle-btn"]';
+    this.resetFiltersButton = '[data-test="traces-search-bar-reset-filters-btn"]';
+    this.sqlModeToggle = '[data-test="logs-search-bar-sql-mode-toggle-btn"]';
```
[the remark this answers is not in the document]
Comment (priyabrata-stack) [reply tests/ui-testing/pages/tracesPages/tracesPage.js:21]: No

**Comment by priyabrata-stack** [comment]
@greptile
